sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -529,69 +603,178 @@ myparams::joint_matrix_b<sub_group, layout::row_major> sub_b;
 myparams::joint_matrix_accumulator<sub_group> sub_c;
 
 ```
+==== Runtime Query
+This provides a more general query interface with information about
+sizes and types that are supported by a specific matrix
+implementation. This is needed to avoid padding by the user, for
+tuning, and efficient code generation if used by a library.
```

Comment:
I'd recommend changing this introduction a bit.  This query isn't "more general".  The main difference is that the information isn't `constexpr` and the application doesn't need to know the type of device on which it will run.  The last sentence about "padding", etc. doesn't seem specific to this query API.  I'd say something like:

> The runtime query does not require the application to hard-code a specific device type, but it also returns values that are not `constexpr`.  It provides similar information as the compile time query API via an extended device information descriptor.